### ISSUE TITLE
`SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data` when submitting form

### ISSUE THREAD

**Issue by jannisborgers** [verbb/formie#1546]
### Describe the bug

The forms on one of my client’s sites stopped working, but only in the live environment. I had hooked into the `onAfterFormieEvaulateConditions`, `onBeforeFormieSubmit` and `onFormieValidate` events to modify the behavior and submit conditionally. Today I discovered I had accidentally done that to all forms, causing submissions on forms that don’t need the hooks to fail silently.

It’s all AJAX-forms using `{{ craft.formie.renderForm() }}`, passing the event listeners on the form via the options. I then modified the form to only hook into the events when using the matching form. 

## Contact form fails but goes through

The simple contact form goes through, emails are sent correctly, but once I submit the form, an alert appears above the form:

```
Unable to parse response `SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data`.
```

The Formie logs only show `Submission triggered for contact` (or other form), listing the form data, which looks okay, and no errors.

## Multi-page form breaks on front end but goes through as well

On the form which hooks into Formie events, the same front end error is thrown when it validates the form after submitting on page 1 of 2, which breaks that form: It doesn't proceed to page 2.

The logs now additionally show errors because fields on page 2 are empty:

```
2023-08-30 16:59:28 [ERROR] Couldn’t save submission due to errors - {"vorname":["Vorname darf nicht leer sein."],"eMail":["E-Mail darf nicht leer sein."],"strasse":["Straße darf nicht leer sein."],"plz1":["PLZ darf nicht leer sein."], … etc.
```

When I manually click onto page 2, no errors are shown, and I can continue filling the form and submit it successfully. The error persists though, and I don’t see a success confirmation after submitting from page 2.

In my local machine, it all works fine, no errors thrown, custom JS validation working fine…

Can I debug this somehow? I have no idea why or from where that error is thrown, but given that it’s only happening live, I suspect that it’s something inside the Formie plugin and not my JS-code…

### Steps to reproduce

?

### Form settings

- Multi-page form: Yes
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: renderForm()

### Craft CMS version

4.4.14

### Plugin version

2.0.32

### Multi-site?

No

**Comment by jannisborgers** [comment]
The weirdness continues:

In a new test, the error on the single-page contact form is now this:

```
Unable to parse response `SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data`.
[{"type":"paragraph","attrs":{"textAlign":"start"}}]
```

Still submits fine, but users get this message on the front end instead of a success message.

**Comment by jannisborgers** [comment]
So that second line is the error message that I can provide in the Form behavior tab. If it’s empty, the above happens. If it’s filled in, I get the text instead:

```
Unable to parse response `SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data`.
Test
```

**Comment by engram-design** [comment]
The JSON error means that a non-JSON response was received from the server. Check the response preview in your Network tab in Chrome developer tools to see what's being returned. It's likely this might be a HTML page of some sort.

What's interesting is that in your last two tests, you are receiving `[{"type":"paragraph","attrs":{"textAlign":"start"}}]` and `Test` as the response back from the server.

The first text is the serialized schema for a rich text editor, which would be the "success message", but why it's serialized and not HTML is strange. The second response of `Test` seems like you might be doing something on your end?

Are there any JS errors in the form builder when saving the form? Any front-end JS errors?

I know some people use the Cloudflare RocketLoader which does mess up Formie's JS.

**Comment by jannisborgers** [comment]
Hi @engram-design — thanks for the quick response.

About the texts: You are right that `Test` is the message I put into the field. And if it’s empty, it’s the serialized schema. I’m not sure if this is a bug (?) or has something to do with my site.

And about the error: No front-end JS errors are logged, and I can successfully save the forms without error in the control panel.

Thanks for suggesting debugging the XHR-request (😬). Spam is returned by the server, that’s why `JSON.parse` fails:

```
<center style="position:absolute;left:-16422px"><a href="https://www.fakerolex.it/">more info</a> convey a certain image: active.a lot of works out take too much time to end up being area very comfortable about <a href="https://www.replicafendiwatches.com/product/fendi-forever-fendi-fow842a2ypf0qg2-womens-19mm-quartz-green-dial/">fendi forever fendi fow842a2ypf0qg2 womens 19mm quartz green dial</a> rolex.</center>{"success":true,"submissionId":14714,"currentPageId":778,"nextPageId":778,"nextPageIndex":1,"totalPages":2,"redirectUrl":"","submitActionMessage":"Anfrage gespeichert.","events":[]}
```
 
Thinking maybe it’s an XSS vulnerability (see this [article](https://security.snyk.io/vuln/SNYK-PHP-CRAFTCMSCMS-5660293)), I updated Craft.

The deployment of that update failed on the live server. I found the reason in `bootstrap.php`, there was PHP code adding the above HTML to all requests matching certain request URIs.

The error is resolved for now. I still have no idea how that code ended up there though.
